packages/server/src/GraphQLCache.js

```diff
@@ -53,9 +53,17 @@ const MAX_READS = 200;
 
 export async function getGraphQLCache(
   configDir: Uri,
+  extensions: ?Array<GraphQLConfig>,
 ): Promise<GraphQLCacheInterface> {
-  const graphQLConfig = await getGraphQLConfig(configDir);
-  return new GraphQLCache(configDir, graphQLConfig);
+  let graphQLConfig = await getGraphQLConfig(configDir);
+  if (extensions && extensions.length > 0) {
+    await Promise.all(
```

Comment:
Promise.all executed in parallel to the order of applying extension is random.
That would lead to race conditions.
You should make applying for extensions sequential process.